vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/scatter_plotly.tsx

```diff
@@ -0,0 +1,232 @@
+// Input component that takes nested object
+//   and shows the keys one level at a time.
+// Returns the last "Leaf" that the user selects.
+import React, {useCallback, useMemo} from 'react';
+import * as _ from 'lodash';
+
+import {DataEnvelope, get_keys, get_vals, nulled_vals, WithInputParams} from './input_types';
+import TextInput from 'etna-js/components/inputs/text_input';
+import { useSetsDefault } from './useSetsDefault';
+import { Maybe, some } from '../../../../selectors/maybe';
+import DropdownAutocomplete from 'etna-js/components/inputs/dropdown_autocomplete';
+import { Button, Slider } from '@material-ui/core';
+import { pick } from 'lodash';
+import NestedSelectAutocompleteInput from './nested_select_autocomplete';
+
+/*
+This input is closely tied to archimedes/functions/plotting/scatter_plotly.
+
+Major design notes:
+- It will have a set of advanced options that are shown/hidden via a toggle button
+- It will also allow workflow-designers to hide controls for any inputs that are set internally by the workflow. (Ex: x_by & y_by for a UMAP)
+
+Input structure:
+  (Minimal = Either of 'data_frame' or 'data_options')
+  'data_options': dictionary of options for '*_by' (ex: 'x_by', 'color_by') inputs where keys are the primary options and None vs [] values indicates what secondary options there may be if there are indeed secondary options (such as for genes!)
+  'data_frame': A hash whose keys will be used as options if 'data_options' is not set
+  'hide': An  array (python list). The names of scatter_plotly inputs which should not be shown in the current widget render.  Hidden inputs should be set elsewhere.  Perhaps they are hardset by the workflow so a user has no choice
+
+Output Structure:
+  dictionary of scatter_plotly inputs-name (key) + value pairs.
+*/
+
+const defaults: DataEnvelope<any> = {
+  'x_by': null,
+  'y_by': null,
+  'color_by': null,
+  'size': 5,
+  'plot_title': 'make',
+  'legend_title': 'make',
+  'xlab': 'make',
+  'ylab': 'make',
+  'color_order': 'increasing',
+  'order_when_continuous_color': false
+};
+
+const remove_hidden = (values: DataEnvelope<any>, hide: string[] | null | undefined) => {
```

Comment:
Same as before, I highly recommend doing a shallow copy.  `values = {...values}`.  This ensures the deletes do not break referential integrity.